mmpose/datasets/datasets/body3d/body3d_h36m_dataset.py

```diff
@@ -0,0 +1,253 @@
+        ann_info = {}
+        ann_info['flip_pairs'] = [[1, 4], [2, 5], [3, 6], [11, 14], [12, 15],
+                                  [13, 16]]
+        ann_info['upper_body_ids'] = (0, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16)
+        ann_info['lower_body_ids'] = (1, 2, 3, 4, 5, 6)
+        ann_info['use_different_joint_weights'] = False
+
+        self.ann_info.update(ann_info)
+
+    def load_annotation(self):
+        data_info = super().load_annotation()
+
+        # convert 3D joints from original 24-keypoint to standard 17-keypoint
+        assert data_info['joint_3d'].shape[1] == self.JOINT_NUM_GLOBAL
+        data_info['joint_3d'] = data_info['joint_3d'][:, self.JOINT_IDX_GLOBAL]
+
+        # get 2D joints
+        if self.joint_2d_src == 'gt':
+            assert data_info['joint_2d'].shape[1] == self.JOINT_NUM_GLOBAL
+            data_info['joint_2d'] = data_info['joint_2d'][:, self.
+                                                          JOINT_IDX_GLOBAL]
+        elif self.joint_2d_src == 'detection':
+            data_info['joint_2d'] = self._load_joint_2d_detection(
+                self.joint_2d_det_file)
+        elif self.joint_2d_src == 'pipeline':
+            pass
+        else:
+            raise NotImplementedError(
+                f'Unhandled joint_2d_src option {self.joint_2d_src}')
+
+        return data_info
+
+    @staticmethod
+    def _parse_h36m_imgname(imgname):
+        """Parse imgname to get information of subject, action and camera.
+
+        A typical h36m image filename is like:
+        S1_Directions_1.54138969_000001.jpg
+        """
+        subj, rest = osp.basename(imgname).split('_', 1)
+        action, rest = rest.split('.', 1)
+        camera, rest = rest.split('_', 1)
+
+        return subj, action, camera
+
+    def build_sample_indices(self):
+        """Split original videos into sequences and build frame indices.
+
+        This method overrides the default one in the base class.
+        """
+
+        # Group frames into videos. Assume that self.data_info is
+        # chronological.
+        video_frames = defaultdict(list)
+        for idx, imgname in enumerate(self.data_info['imgnames']):
+            subj, action, camera = self._parse_h36m_imgname(imgname)
+            video_frames[(subj, action, camera)].append(idx)
+
+        # build sample indices
+        sample_indices = []
+        _len = (self.seq_len - 1) * self.seq_frame_interval + 1
+        _step = self.seq_frame_interval
+        for _, _indices in sorted(video_frames.items()):
+            n_frame = len(_indices)
+            seqs_from_video = [
+                _indices[i:(i + _len):_step]
+                for i in range(0, n_frame - _len + 1)
+            ]
+            sample_indices.extend(seqs_from_video)
+
+        return sample_indices
+
+    def evaluate(self, outputs, res_folder, metric='joint_error', **kwargs):
+        metrics = metric if isinstance(metric, list) else [metric]
+
```

Comment:
Try to bound the kwargs, like

assert len(kwargs) == 0

Reply:
I thought that the purpose of the "kwargs" here is to allow arbitrary args passed in but not used. It seems that "kwargs" is not bounded in other dataset classes' evaluate method.

Reply:
Yeah this is the source of bugs. It is dangerous to omit arbitrary args. I am advocating to bound the kwargs, and it is still in-progress.

Reply:
Then maybe just remove **kwargs from the argument list?